src/main/java/dev/openfeature/sdk/OpenFeatureAPI.java

```diff
@@ -100,15 +96,18 @@ public Client getClient(String domain) {
      * Multiple clients can be used to segment feature flag configuration.
      * If there is already a provider bound to this domain, this provider will be used.
      * Otherwise, the default provider is used until a provider is assigned to that domain.
-     * 
-     * @param domain a identifier which logically binds clients with providers
+     *
+     * @param domain  a identifier which logically binds clients with providers
      * @param version a version identifier
      * @return a new client instance
      */
     public Client getClient(String domain, String version) {
-        return new OpenFeatureClient(this,
+        return new OpenFeatureClient(
+                () -> providerRepository.getProvider(domain),
```

Comment:
Inside the `ProviderRespository`, you will have to modify [this private method](https://github.com/open-feature/java-sdk/blob/64082617fade3fa7d647302cc0c5b698f7038d81/src/main/java/dev/openfeature/sdk/ProviderRepository.java#L129) to call the initialize function on the provider if it's not yet initialized or in the process of being initialized, then update a new package-private status member indicating readiness.